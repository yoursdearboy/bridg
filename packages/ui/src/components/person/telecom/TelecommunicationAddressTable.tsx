import { Box, Group, Modal, Table } from "@mantine/core";
import { useDisclosure, useHover } from "@mantine/hooks";
import { IconPencil, IconX } from "@tabler/icons-react";
import { useMutation } from "@tanstack/react-query";
import type { TelecommunicationAddress } from "api-ts";
import { t } from "i18next";
import api from "@/api";
import { EditTelecommunicationAddressForm } from "./EditTelecommunicationAddressForm";
import icons from "./icons";

interface TelecommunicationAddressTableRowWrapperProps {
  personId: string;
  telecommunication_address: TelecommunicationAddress;
}

const TelecommunicationAddressTableRowWrapper = ({
  personId,
  telecommunication_address,
}: TelecommunicationAddressTableRowWrapperProps) => {
  const mutation = useMutation({
    mutationKey: [
      "person",
      personId,
      "telecommunication_address",
      telecommunication_address.id,
    ],
    mutationFn: () =>
      api.persons.deletePersonsPersonIdTelecommunicationAddressesAddressIdDelete(
        {
          personId,
          addressId: telecommunication_address.id,
        }
      ),
  });

  return (
    <TelecommunicationAddressTableRow
      telecommunication_address={telecommunication_address}
      personId={personId}
      onDelete={() => mutation.mutate()}
    />
  );
};

interface TelecommunicationAddressTableRowProps {
  telecommunication_address: TelecommunicationAddress;
  personId: string;
  onDelete: (name: TelecommunicationAddress) => void;
}

const TelecommunicationAddressTableRow = ({
  telecommunication_address,
  personId,
  onDelete,
}: TelecommunicationAddressTableRowProps) => {
  const { hovered, ref } = useHover();
  const [opened, { open, close }] = useDisclosure(false);
  const handleEdit = () => {
    open();
  };
  const handleDelete = () => {
    if (window.confirm("Удалить выбранное значение?")) {
      onDelete(telecommunication_address);
    }
  };

  const Icon = telecommunication_address.scheme
    ? icons[telecommunication_address.scheme]
    : () => <div></div>;
  return (
    <>
      <Table.Tr ref={ref}>
        <Table.Td width={30} valign="middle">
          <Icon size={16} strokeWidth={2} display="block" />
        </Table.Td>
        <Table.Td>{telecommunication_address.address}</Table.Td>
        <Table.Td width={60}>
          {hovered && (
            <Group gap={8}>
              <IconPencil size={16} color="green" onClick={handleEdit} />
              <IconX size={16} color="red" onClick={handleDelete} />
            </Group>
          )}
        </Table.Td>
      </Table.Tr>
      <Modal opened={opened} onClose={close} title={t("edit")} size="lg">
        <EditTelecommunicationAddressForm
          personId={personId}
          telecommunication_address={telecommunication_address}
          onCancel={close}
          onSuccess={() => close()}
        />
      </Modal>
    </>
  );
};

interface TelecommunicationAddressTableProps {
  personId: string;
  telecom_addresses: TelecommunicationAddress[];
}

export const TelecommunicationAddressTable = ({
  personId,
  telecom_addresses,
}: TelecommunicationAddressTableProps) => {
  return (
    <Box>
      <Table highlightOnHover>
        <Table.Tbody>
          {telecom_addresses.length === 0 ? (
            <Table.Tr>
              <Table.Td px={0} style={{ textAlign: "center" }}>
                {t("nodata")}
              </Table.Td>
            </Table.Tr>
          ) : (
            telecom_addresses.map((name) => (
              <TelecommunicationAddressTableRowWrapper
                key={name.id}
                personId={personId}
                telecommunication_address={name}
              />
            ))
          )}
        </Table.Tbody>
      </Table>
    </Box>
  );
};
