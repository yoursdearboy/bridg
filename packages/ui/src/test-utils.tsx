/* eslint-disable react-refresh/only-export-components */
import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  type AnyRoute,
  createRootRoute,
  createRootRouteWithContext,
  createRoute,
  createRouter,
  RouterProvider,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";
import type { PropsWithChildren } from "react";

const queryClient = new QueryClient();

const App = ({ children }: React.PropsWithChildren) => {
  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params, props }: { params?: object; props?: PropsWithChildren } = {}
) => {
  const rootRoute = createRootRouteWithContext()();
  let routeConfig = {
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  };
  if (props?.children)
    routeConfig = { ...routeConfig, component: () => props.children };

  const testRoute = createRoute(routeConfig);
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
    context: {
      queryClient,
    },
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};

export const renderComponent = (component: React.ReactNode) =>
  render(<App>{component}</App>);

export const EmptyRouterProvider = (props: PropsWithChildren) => {
  const rootRoute = createRootRoute({
    component: () => props.children,
  });

  return renderRoute(rootRoute, { props: { children: props.children } });
};
