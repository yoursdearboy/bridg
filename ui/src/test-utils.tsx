import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  createRootRoute,
  createRoute,
  createRouter,
  RouterProvider,
  type AnyRoute,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";

// eslint-disable-next-line react-refresh/only-export-components
const App = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params }: { params?: object } = {}
) => {
  const rootRoute = createRootRoute();
  const testRoute = createRoute({
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  });
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};
