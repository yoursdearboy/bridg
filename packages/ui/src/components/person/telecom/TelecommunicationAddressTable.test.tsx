import { describe, expect, it } from "vitest";
import { renderComponent } from "@/test-utils";
import { TelecommunicationAddressTable } from "./TelecommunicationAddressTable";

describe("TelecommunicationAddressTable", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderComponent(
      <TelecommunicationAddressTable
        telecom_addresses={[
          {
            id: "1",
            label: "ftp Pizza",
          },
          {
            id: "2",
            label: "http Pineapple",
          },
        ]}
        personId="1"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
