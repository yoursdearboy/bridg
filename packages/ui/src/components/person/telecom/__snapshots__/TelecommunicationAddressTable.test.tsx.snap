// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TelecommunicationAddressTable > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
  >
    <table
      class="m_b23fa0ef mantine-Table-table"
      style="--table-vertical-spacing: calc(0.4375rem * var(--mantine-scale));"
    >
      <tbody
        class="m_b2404537 mantine-Table-tbody"
      >
        <tr
          class="m_4e7aa4fd mantine-Table-tr"
          data-hover="true"
          data-with-row-border="true"
        >
          <td
            class="m_4e7aa4ef mantine-Table-td"
            valign="middle"
            width="30"
          >
            <div />
          </td>
          <td
            class="m_4e7aa4ef mantine-Table-td"
          />
          <td
            class="m_4e7aa4ef mantine-Table-td"
            width="60"
          />
        </tr>
        <tr
          class="m_4e7aa4fd mantine-Table-tr"
          data-hover="true"
          data-with-row-border="true"
        >
          <td
            class="m_4e7aa4ef mantine-Table-td"
            valign="middle"
            width="30"
          >
            <div />
          </td>
          <td
            class="m_4e7aa4ef mantine-Table-td"
          />
          <td
            class="m_4e7aa4ef mantine-Table-td"
            width="60"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
